servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -456,25 +378,48 @@ public void completeFile(long fileId)
         }
       }
 
-      completeFileInternal(fileInode.getBlockIds(), fileId, fileLength, false, opTimeMs);
+      // If the file is both persisted and stored in memory, the memory footprint and the UFS file
+      // length should match.
```

Comment:
Maybe update this comment? I'm not sure if the check is even necessary anymore since it seems so loose.